### PR TITLE
OODT-918 - Prerequisite workflow condition to check ordering of Resource Manager jobs

### PR DESCRIPTION
**Pull request by karanjeets**
Created a new workflow condition that enable us to check for prerequisite jobs scheduled in Resource Manager.

**Assumption**
Job names and Resource Manager Url is provided through configuration file.

**JIRA Ticket**
OODT-918

### CODE REVIEW

File: workflow/src/main/java/org/apache/oodt/cas/workflow/examples/PrerequisiteCondition.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	/* (non-Javadoc)
+	 * @see org.apache.oodt.cas.workflow.structs.WorkflowConditionInstance#evaluate(org.apache.oodt.cas.metadata.Metadata)
+	 */
+    public boolean evaluate(Metadata metadata, WorkflowConditionConfiguration config) {
+		
+		String[] jobs = (config.getProperty("JOBS")).split(",");
+		String resourceManagerUrl = config.getProperty("RESMGR_URL");
+		boolean flag = true;
+
+		System.out.println(new Date() + " PrerequisiteCondition: Jobs: "+ Arrays.toString(jobs));
```
Comment (chrismattmann) [inline workflow/src/main/java/org/apache/oodt/cas/workflow/examples/PrerequisiteCondition.java]: Change to LOG.info

File: workflow/src/main/java/org/apache/oodt/cas/workflow/examples/PrerequisiteCondition.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		System.out.println(new Date() + " PrerequisiteCondition: Resource Manager: "+ resourceManagerUrl);
```
Comment (chrismattmann) [inline workflow/src/main/java/org/apache/oodt/cas/workflow/examples/PrerequisiteCondition.java]: LOG.info()

File: workflow/src/main/java/org/apache/oodt/cas/workflow/examples/PrerequisiteCondition.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+/**
+ * @author singhk
+ * @version $Revision$
+ *
+ * <p>A Simple condition that evaluates to false as many times as the prerequisite job in Resource Manager
+ * isn't completed. After that, the condition returns
+ * true.</p>
+ *
+ */
+public class PrerequisiteCondition implements WorkflowConditionInstance {
```
Comment (chrismattmann) [inline workflow/src/main/java/org/apache/oodt/cas/workflow/examples/PrerequisiteCondition.java]: maybe rename the class to ResmgrJobCondition?

**Comment by karanjeets** [comment]
@chrismattmann  Done. :)

**Comment by karanjeets** [comment]
@chrismattmann Since, we have moved DRAT to OODT-0.10, can ResmgrJobCondition be available in that ?

**Comment by chrismattmann** [comment]
@karanjeets we should just release 0.11 and make sure that DRAT is compatible with it. make sense?

**Comment by karanjeets** [comment]
okay. got it :)

**Comment by buggtb** [comment]
Guys (Chris) I don't think you should ship 0.11 with OODT-781 still open, I'll see if I can find some time over the next few days to look at it, but not being able to delete products is pretty bad I would say!

Tom

**Comment by chrismattmann** [comment]
got it @buggtb agreed. Let's try and get [OODT-781](http://issues.apache.org/jira/browse/OODT-781) handled as part of 0.11.

**Pull request closed without merging**